lib/routes/account.js

```diff
@@ -2169,6 +2170,22 @@ module.exports = (
             })
         }
 
+        function deleteSecondaryEmailIfUnverified() {
+          return db.getSecondaryEmail(email)
+            .then((secondaryEmailRecord) => {
+              // Only delete secondary email if it is unverified and does not belong
+              // to the current user.
+              if (! secondaryEmailRecord.isVerified && secondaryEmailRecord.uid !== uid.toString('hex')) {
```

Comment:
Hm, so the `uid` we get from the db on a `sessionToken` is a `Buffer`, but the `uid` on `secondaryEmailRecord` is a string?